azurerm/resource_arm_network_security_group.go

```diff
@@ -53,30 +53,67 @@ func resourceArmNetworkSecurityGroup() *schema.Resource {
 						},
 
 						"protocol": {
-							Type:         schema.TypeString,
-							Required:     true,
-							ValidateFunc: validateNetworkSecurityRuleProtocol,
-							StateFunc:    ignoreCaseStateFunc,
+							Type:             schema.TypeString,
+							Required:         true,
+							ValidateFunc:     validateNetworkSecurityRuleProtocol,
+							StateFunc:        ignoreCaseStateFunc,
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
 						},
 
 						"source_port_range": {
-							Type:     schema.TypeString,
-							Required: true,
+							Type:          schema.TypeString,
+							Optional:      true,
+							ConflictsWith: []string{"security_rule.source_port_ranges"},
```

Comment:
so as this is a list, in order for this to work we'd need to include the index of the item in the list e.g. `security_rule.0.source_port_ranges` - unfortunately since this is an unbounded list there's currently no easy way to access the current index - as such I don't think there's a way to make this work at the moment, unfortunately (although, some changes coming in the future may make that possible.

As such could we remove the `ConflictsWith` for the moment? We can add an info box to the documentation explaining this